frontend/src/metabase/components/form/CustomForm/types.ts

```diff
@@ -4,16 +4,16 @@ import {
   DefaultFieldValue,
   FieldValues,
   FormFieldDefinition,
-  FormField,
   FormObject,
+  NestedFormField,
 } from "metabase-types/forms";
 
 export interface BaseFormProps {
   formKey?: string;
   formName: string;
   formObject: FormObject;
 
-  fields: FormField[];
+  fields: NestedFormField;
```

Comment:
Fix the `fields` type. It should be an object, not an array.

#### Example: database form

![image](https://user-images.githubusercontent.com/1937582/170470023-b6930b35-6201-4196-9b8f-39c054ee6d7d.png)